aspnetcore/fundamentals/servers/kestrel/endpoints.md

```diff
@@ -20,7 +20,8 @@ uid: fundamentals/servers/kestrel/endpoints
 Kestrel endpoints provide the infrastructure for listening to incoming requests and routing them to the appropriate middleware. The combination of an address and a protocol defines an endpoint.
 
 * The address specifies the network interface that the server listens on for incoming requests, such as a TCP port.
-* The protocol defines the communication standard between the client and server, such as whether the endpoint is secured with HTTPS, and the HTTP version.
+* The scheme specifies if the endpoint is secured with HTTPS.
+* The protocol specifies the communication standard between the client and server, such as HTTP/1.1, HTTP/2, or HTTP/3.
```

Comment:
Is scheme the right word? If you're configuring endpoints with code you'd call `UseHttps`. It's only URLs and config that set https via a scheme.

Reply:
In some places, such as [here]( https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-8.0&branch=pr-en-us-29605#url-formats), we imply that protocol means HTTP vs. HTTPS. In other places it’s HTTP/1 vs. HTTP/2, etc., or TLS1.1 vs. TLS1.2 etc. But the introduction to this bulleted list uses "protocol" loosely to carry all these meanings. So would it also be more accurate to replace “The combination of an address and a protocol defines an endpoint.” with “The combination of an address and a scheme defines an endpoint.”? Or "The combination of an address, a scheme, and one or two protocols defines an endpoint."?

Reply:
The uri prefix (scheme, address, port) define an endpoint. Endpoints support one or more HTTP protocol versions.

Reply:
I especially wanted to split them out because the next list does too.

Reply:
> Is scheme the right word? If you're configuring endpoints with code you'd call `UseHttps`. It's only URLs and config that set https via a scheme.

It's the closest we've got that is distinguishable from http protocol versions.

Reply:
Hmm, maybe your right, the address and the port really do define the endpoint. An endpoint may have encryption enabled via scheme or UseHttps, and it will support one or more HTTP protocol versions.

Reply:
I updated the bullet point to say:

> An endpoint can be secured with HTTPS using the `HTTPS` scheme or `UseHttps` method.

What do you think?

Reply:
Maybe "`https` url scheme"?